### PR TITLE
Update About Template Functions topic

### PR DESCRIPTION
**Pull request by paigecalvert**
https://app.shortcut.com/replicated/story/60044/move-using-template-functions-topic-to-references-section

Changes:
- Removed the Using Template Functions topic, removed xrefs, added redirects, removed it from the TOC
- Moved content from Using Template Functions to the About Template Functions topic in the References section: https://deploy-preview-709--replicated-docs.netlify.app/reference/template-functions-about
- Updated the content/organization in the About Template Functions topic and added some partials to be able to render YAML in the Syntax tables
- Moved the TLS Certs example to the Config Context section, since it's specific to the config context functions (I did some light editing on this example too. Though, this might need to eventually be turned into a real procedure and/or moved to a section about creating tls certs/keys, but that felt out of scope for this story): https://deploy-preview-709--replicated-docs.netlify.app/reference/template-functions-config-context#examples

### CODE REVIEW

File: docs/reference/template-functions-config-context.md [hunk 1]
```diff
@@ -248,3 +248,58 @@ func HasLocalRegistry() bool
...
+### Example: Using Variables to Generate TLS Certificates and Keys {#tls-cert-example}
+
+For template functions that Replicated evaluates at the same time, you can assign the result returned by a template function to a variable in a manifest file, then use the variable in another template function.
+
+Replicated evaluates and templates each item in the Config custom resource separately. Template functions in the Config custom resource do not have access to variables defined in other configuration fields.
+
+As a workaround, you can create a configuration item with `hidden` set to `true` to evaluate complex templates and render the results.
+Then, you can access the result using the [`ConfigOption`](#configoption) function.
+
+The following Config custom resource example demonstrates how to generate a CA, a cert, and a key using [Sprig](http://masterminds.github.io/sprig/) functions. It also shows how to use a hidden `tls_json` configuration field that contains all the generated values in JSON format, and then reference the `tls_json` field in other configuration fields.
```
Comment (jonquil2002) [inline docs/reference/template-functions-config-context.md:265]: s/ how to generate a certificate authority (CA), a certificate,

I think CA is certificate authority...but anyway, the idea is to define the acronym and spell out certificate.

**Review by jonquil2002 — DISMISSED** [review]
lgtm - just one minor nit.

File: docs/reference/template-functions-config-context.md (hunk 1 above)
Comment (paigecalvert) [reply docs/reference/template-functions-config-context.md:265]: Great catch

**Review by jonquil2002 — APPROVED** [review]
lgtm

**Pull request merged**